File: RedisConCrud/demo2/src/main/java/com/example/demo/service/UserServiceImp.java
package com.example.demo.service;

import com.example.demo.entity.User;
import com.example.demo.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import java.util.List;


@Service
public class UserServiceImp implements UserService{

    @Autowired
    UserRepository userRepository;

    @Override
    public User addUser(User user) {
        return userRepository.save(user);
    }

    @Override
    public User updateUser(User user, String name ) {
        User newUser = userRepository.findUsersByName(name);
        newUser.setName(user.getName());
        newUser.setNickName(user.getNickName());
        newUser.setAge(user.getAge());
        return userRepository.save(newUser);
    }

    @Transactional
    @Override
    public void deleteUser(User user) {
        userRepository.delete(user);
    }

    public User getUserByNickName(String nickname){
        return userRepository.findUsersByNickName(nickname);
    }

    public User getUserByName(String name){
        return userRepository.findUsersByName(name);
    }

    public List<User> getAll(){
        return userRepository.findAll();
    }
}

File: kotlin/kotlinSpring/src/main/kotlin/com/joanna/kotlindemo/service/UserServiceImp.kt
package com.joanna.kotlindemo.service

import com.joanna.kotlindemo.entity.User
import com.joanna.kotlindemo.repository.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserServiceImp : UserService {

    @Autowired
    var userRepo: UserRepository? = null

    override fun addUser(user: User): User {
        return userRepo!!.save(user)
    }

    override fun updateUser(user: User, name: String): User {
        val newUser = userRepo!!.findUsersByName(name!!)
        newUser.name=user.name
        //newUser.setName(user.getName())
        newUser.nickName=user.nickName
        //newUser.setNickName(user.getNickName())
        newUser.age=user.age
        //newUser.setAge(user.getAge())
        return userRepo!!.save(newUser)
    }

    @Transactional
    override fun deleteUser(user: User) {
        userRepo!!.delete(user)
    }


    fun getUserByNickName(nickname: String?): User {
        return userRepo!!.findUsersByNickName(nickname!!)
    }

    fun getUserByName(name: String?): User {
        return userRepo!!.findUsersByName(name!!)
    }

    fun getAll(): List<User?> {
        return userRepo!!.findAll()
    }
}
